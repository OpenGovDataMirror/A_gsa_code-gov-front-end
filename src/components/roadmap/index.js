import RoadmapContainer from './roadmap.container'

export default RoadmapContainer
