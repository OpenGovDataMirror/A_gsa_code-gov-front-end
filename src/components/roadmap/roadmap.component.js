import React from 'react'
import Breadcrumbs from 'components/breadcrumbs'
import SiteBanner from 'components/site-banner'
import { normalize, refreshView } from 'utils/other'
import { map } from '@code.gov/cautious'

export const Row = ({ todo }) => {
  let iconClass = 'icon'
  let taskStatus = ''
  const status = normalize(todo.status)
  if (status === 'released') {
    iconClass += ' icon-ok text-green'
    taskStatus = 'released'
  } else if (status === 'in progress') {
    iconClass += ' icon-circle text-accent-warm-light'
    taskStatus = 'in progress'
  } else {
    taskStatus = 'no status'
  }

  return (
    <li className="margin-bottom-0">
      <div className="grid-row">
        <div className="grid-col-1 padding-top-1 padding-left-2">
          <i className={iconClass} aria-label={taskStatus} />
        </div>
        <div className="grid-col-11 padding-y-1 padding-x-1 font-body-3xs text-base-dark padding-left-3">
          {todo.name}
        </div>
      </div>
    </li>
  )
}

export const Column = ({ phase, todos }) => (
  <div className="tablet:grid-col">
    <div>
      <h3 className="bg-base-lighter padding-y-2 text-center text-bold text-base-darker border-base-lighter font-body-md radius-top-lg border-1px border-bottom-0">
        {phase}
      </h3>
      <ul className="radius-bottom-lg margin-top-0 padding-y-1 padding-x-05 border-1px border-top-0 border-base-lighter">
        {map(todos, todo => (
          <Row key={todo.name} todo={todo} />
        ))}
      </ul>
    </div>
  </div>
)

export default class Roadmap extends React.Component {
  componentDidMount() {
    refreshView()
  }

  get overview() {
    return (
      <div className="margin-top-1">
        <h2>Overview</h2>
        {map(this.props.overview, paragraph => (
          <p
            className="maxw-none"
            key={paragraph}
            dangerouslySetInnerHTML={{ __html: paragraph }}
          />
        ))}
      </div>
    )
  }

  render() {
    return (
      <main id="main-content">
        <SiteBanner title="Roadmap" />
        <Breadcrumbs crumbs={[{ text: 'Home', to: '/' }, { text: 'Roadmap' }]} />

        <div className="grid-container">
          {this.overview}

          <div>
            <h2 className="margin-top-4">Roadmap</h2>
          </div>

          <section className="grid-container padding-0">
            <div className="grid-row grid-gap padding-top-3">
              <Column phase="Near-term" todos={this.props.near} />
              <Column phase="Mid-term" todos={this.props.mid} />
              <Column phase="Long-term" todos={this.props.long} />
            </div>
          </section>

          <section
            className="grid-container margin-top-205 margin-bottom-5 padding-0"
            aria-hidden="true"
          >
            <div className="grid-row border-0 padding-bottom-0">
              <i className="icon icon-ok text-green" />
              <span className="border-0 font-body-3xs padding-left-1 padding-bottom-0 text-base-dark padding-top-1px">
                Released
              </span>
            </div>
            <div className="grid-row border-0 padding-top-1">
              <i className="icon icon-circle text-accent-warm-light" />
              <span className="border-0 font-body-3xs padding-left-1 text-base-dark padding-top-1px">
                In Progress
              </span>
            </div>
          </section>

          <div className="margin-y-3">
            <h2>Disclaimer</h2>
            <p className="maxw-none">{this.props.disclaimer}</p>
          </div>
        </div>
      </main>
    )
  }
}
