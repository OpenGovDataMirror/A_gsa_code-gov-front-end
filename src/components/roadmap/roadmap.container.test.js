import { mapStateToProps } from 'components/roadmap/roadmap.container'

describe('containers - Roadmap', () => {
  describe('mapStateToProps', () => {
    it('should return the correct properties', () => {
      expect(mapStateToProps()).toMatchSnapshot()
    })
  })
})
