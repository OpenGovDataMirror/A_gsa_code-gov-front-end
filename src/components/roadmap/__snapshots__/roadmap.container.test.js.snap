// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers - Roadmap mapStateToProps should return the correct properties 1`] = `
Object {
  "disclaimer": "This Development Roadmap (Roadmap) is for informational purposes only. The Code.gov team will update the Roadmap as frequently as possible but it is subject to change at any time. The information presented by the Roadmap should not be used for planning purposes. The information presented in the Roadmap does not represent a solicitation or similar contract vehicle. All items in the Roadmap will be developed and released at the sole discretion of Code.gov and the U.S. General Services Administration.",
  "long": Array [
    Object {
      "name": "Metadata schema update",
      "status": "null",
    },
    Object {
      "name": "New validator tool w/ new schema",
      "status": "null",
    },
    Object {
      "name": "Agency compliance update (replaces IDC metrics)",
      "status": "null",
    },
    Object {
      "name": "Website search capabilities, integration w/ Search.gov",
      "status": "null",
    },
    Object {
      "name": "Public engagement metrics (Phase 2: Site deployment)",
      "status": "null",
    },
  ],
  "mid": Array [
    Object {
      "name": "Complete USWDS 2.0 migration (80% complete)",
      "status": "In Progress",
    },
    Object {
      "name": "Public engagement metrics (Phase 1: Pre-site deployment)",
      "status": "In Progress",
    },
    Object {
      "name": "Explore new site design ideas",
      "status": "null",
    },
    Object {
      "name": "Find new solution for code quality reviews",
      "status": "null",
    },
    Object {
      "name": "Automate sitemap generation",
      "status": "null",
    },
  ],
  "near": Array [
    Object {
      "name": "Build out the Federal Source Code Toolkit",
      "status": "In Progress",
    },
    Object {
      "name": "Migration of web components to React components",
      "status": "In Progress",
    },
    Object {
      "name": "Continue enhancing automated testing across site",
      "status": "In Progress",
    },
    Object {
      "name": "SSP annual review and update (for API ATO)",
      "status": "null",
    },
  ],
  "overview": Array [
    "At Code.gov, we know it’s important to keep an open dialogue with our users. That’s why we’re sharing our plans for the future through this Development Roadmap (aka Roadmap). Hopefully, it provides a better understanding of our priorities for the platform.",
    "The Development Roadmap is organized into three categories – Near-term, Mid-term, and Long-term. Priorities are sorted into roadmap categories based on feedback we receive during user interviews. Dependencies and resources are also factors that affect placement on the roadmap. If you want to join a Code.gov user testing group, please email us at <a href='mailto:code@gsa.gov'>Code@gsa.gov</a>.",
    "Last updated: 4/17/2020",
  ],
}
`
